src/alks.ts

```diff
@@ -664,15 +689,24 @@ namespace ALKS {
      * })
      */
     async createRole(props: CreateRoleProps): Promise<Role> {
-      const results = await this.internalFetch('createRole', props);
-      results.denyArns = results.denyArns.split(',');
-      return pick(results, [
-        'roleArn',
-        'denyArns',
-        'instanceProfileArn',
-        'addedRoleToInstanceProfile',
-        'tags',
-      ]);
+      let roleTypeExists = props.roleType != null;
+      let trustPolicyExists = props.trustPolicy != null;
```

Comment:
This is just a style issue but use `const` when you have a variable whose value won't be conditionally overwritten. Doing this enables compile-time checking and code completion to help you catch when you accidentally try to overwrite a value that you just intended to read from or modify